src/d3a/models/strategy/external_strategies/load.py

```diff
@@ -127,15 +134,17 @@ def _bid(self, payload):
             return
         try:
             arguments = json.loads(payload["data"])
-            assert set(arguments.keys()) == {'price', 'energy', 'transaction_id'}
+            assert all(arg in {'price', 'energy', 'replace_existing'} for arg in arguments.keys())
```

Comment:
And this works? How about the `transaction_id` key?

Reply:
That shouldn't be working, actually. Tests are passing though, which means that something needs to be corrected there too.

Reply:
This is why i added the general comment https://github.com/gridsingularity/d3a/pull/1038#pullrequestreview-582141575 . We currently dont test the external strategies here in the d3a. Thats why the tests are passing.